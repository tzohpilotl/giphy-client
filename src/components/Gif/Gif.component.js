import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { isFavorite } from '../../store/selectors';
import { toggleFavorite } from '../../store/favorites.reducer';
import HeartFilled from './heart-filled.png';
import Heart from './heart.png';
import './Gif.css';

export const Gif = ({ id, isFavorite, gif, toggleFavorite }) => {
  const title = gif.title;
  const imageUrl = gif.images.original.url;

  return (
    <div className="image-section" onClick={() => toggleFavorite(id, gif)}>
      <div className="favorite-icon">
        <img src={isFavorite ? HeartFilled : Heart} alt="Heart" />
      </div>
      <img src={imageUrl} alt={title} />
    </div>
  );
};

const mapStateToProps = (state, ownProps) => ({
  isFavorite: isFavorite(state, ownProps.id),
});

const mapDispatchToProps = dispatch => ({
  toggleFavorite: bindActionCreators(toggleFavorite, dispatch),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Gif);
