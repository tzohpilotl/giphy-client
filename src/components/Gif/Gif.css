.image-section {
  position: relative;
}

.favorite-icon {
  position: absolute;
  top: 0;
  right: 0;
  cursor: pointer;
}