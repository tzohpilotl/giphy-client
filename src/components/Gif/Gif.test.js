import React from 'react';
import { shallow } from 'enzyme';
import { Gif } from './Gif.component';

const info = {
  data: {
    id: 'jj2',
    title: 'My gif',
    images: {
      original: { url: 'http://thegifurl.com' },
    },
  },
};

describe('Spinner component', () => {
  let wrapper = shallow(<Gif id={info.data.id} gif={info.data} />);

  it('should render', () => {
    expect(wrapper.isEmptyRender()).toEqual(false);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have proper attributes', () => {
    const image = wrapper.find('img').at(1);
    const props = image.props();
    expect(props.src).toEqual(info.data.images.original.url);
    expect(props.alt).toEqual(info.data.title);
  });
});
