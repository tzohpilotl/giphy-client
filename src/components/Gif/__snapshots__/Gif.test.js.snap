// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gif
    gif={
      Object {
        "id": "jj2",
        "images": Object {
          "original": Object {
            "url": "http://thegifurl.com",
          },
        },
        "title": "My gif",
      }
    }
    id="jj2"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="favorite-icon"
        >
          <img
            alt="Heart"
            src="heart.png"
          />
        </div>,
        <img
          alt="My gif"
          src="http://thegifurl.com"
        />,
      ],
      "className": "image-section",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Heart"
            src="heart.png"
          />,
          "className": "favorite-icon",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Heart",
            "src": "heart.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "My gif",
          "src": "http://thegifurl.com",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="favorite-icon"
          >
            <img
              alt="Heart"
              src="heart.png"
            />
          </div>,
          <img
            alt="My gif"
            src="http://thegifurl.com"
          />,
        ],
        "className": "image-section",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Heart"
              src="heart.png"
            />,
            "className": "favorite-icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Heart",
              "src": "heart.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "My gif",
            "src": "http://thegifurl.com",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
